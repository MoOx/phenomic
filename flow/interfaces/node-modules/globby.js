declare module "globby" {
  declare type pattern = string | Array<string>;
  declare type async = (
    pattern: pattern,
    options: Object
  ) => Promise<Array<string>>;
  declare var exports: {
    (): async,
    sync: (pattern: pattern, options: Object) => Array<string>
  };
}
