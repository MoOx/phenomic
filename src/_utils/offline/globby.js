// @flow
import globby from "globby"

export default function(
  pattern: Array<string>,
  cwd: string,
  options?: { [key: string]: any }
): Promise {
  return globby(pattern, {
    cwd,
    nodir: true,
    ...options,
  })
}
