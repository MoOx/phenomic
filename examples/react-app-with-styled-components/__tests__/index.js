/* eslint-disable import/no-extraneous-dependencies */
import path from "path";
import fs from "fs";

import globby from "globby";

it("should build example correctly", () => {
  const testFolder = __dirname + "/../dist";
  const files = globby.sync("**/*", {
    cwd: testFolder,
    nodir: true
  });

  // should have html files
  const htmlFiles = files.filter(file => file.endsWith(".html"));
  expect(htmlFiles).toMatchSnapshot();

  const file1 = fs.readFileSync(
    path.join(__dirname, "..", "dist", "index.html"),
    { encoding: "utf8" }
  );
  expect(file1).toContain("Hello World!");
  expect(file1).toContain(
    "font-size:1.5em;text-align:center;color:palevioletred;"
  );

  const file2 = fs.readFileSync(
    path.join(__dirname, "..", "dist", "2", "index.html"),
    { encoding: "utf8" }
  );
  expect(file2).toContain("Hello again!");
  expect(file2).toContain("font-size:2em;text-align:right;color:blue;");
});
