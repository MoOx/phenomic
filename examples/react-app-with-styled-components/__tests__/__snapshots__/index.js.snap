// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should build example correctly 1`] = `
Array [
  "2/index.html",
  "index.html",
]
`;
